consensus/src/round_manager.rs

```diff
@@ -1007,22 +1107,73 @@ impl RoundManager {
                         BlockStage::QC_AGGREGATED,
                     );
                 }
-                self.new_qc_aggregated(qc, vote.author()).await
+                QC_AGGREGATED_FROM_VOTES.inc();
+                let result = self.new_qc_aggregated(qc.clone(), vote.author()).await;
+                if self.onchain_config.order_vote_enabled() {
+                    if result.is_ok() {
+                        let _ = self.broadcast_order_vote(vote, qc.clone()).await;
+                    } else {
+                        warn!(
+                            "OrderVoteBrodcastFailed. Round = {}. Block id = {}",
+                            round,
+                            vote.vote_data().proposed().id()
+                        );
+                    }
+                }
+                result
             },
             VoteReceptionResult::New2ChainTimeoutCertificate(tc) => {
                 self.new_2chain_tc_aggregated(tc).await
             },
             VoteReceptionResult::EchoTimeout(_) if !self.round_state.is_vote_timeout() => {
                 self.process_local_timeout(round).await
             },
-            VoteReceptionResult::VoteAdded(_)
-            | VoteReceptionResult::VoteAddedQCDelayed(_)
+            VoteReceptionResult::VoteAdded(_) => {
+                PROPOSAL_VOTE_ADDED.inc();
+                Ok(())
+            },
+            VoteReceptionResult::VoteAddedQCDelayed(_)
             | VoteReceptionResult::EchoTimeout(_)
             | VoteReceptionResult::DuplicateVote => Ok(()),
             e => Err(anyhow::anyhow!("{:?}", e)),
         }
     }
 
+    async fn process_order_vote_reception_result(
+        &mut self,
+        order_vote: &OrderVote,
+        result: OrderVoteReceptionResult,
+    ) -> anyhow::Result<()> {
+        match result {
+            OrderVoteReceptionResult::NewLedgerInfoWithSignatures(ledger_info_with_signatures) => {
+                if let Some(order_voted_block) = self
+                    .block_store
+                    .get_block(order_vote.ledger_info().consensus_block_id())
+                {
+                    if !order_voted_block.block().is_nil_block() {
+                        observe_block(
+                            order_voted_block.block().timestamp_usecs(),
+                            BlockStage::ORDER_VOTE_QC_CREATED,
+                        );
+                    }
+                }
+                self.new_ordered_cert(
+                    WrappedLedgerInfo::new(VoteData::dummy(), ledger_info_with_signatures),
+                    order_vote.author(),
+                )
+                .await
+            },
```

Comment:
we want to log/have metric when the order cert aggregated but block doesn't exist